rllib/execution/parallel_requests.py

```diff
@@ -182,3 +184,85 @@ def wait_asynchronous_requests(
         remote_fn=None,
     )
     return ready_requests
+
+
+class AsyncRequestsManager:
+    """Manages asynchronous requests to actors."""
+
+    def __init__(
+        self,
+        workers: Union[List[ActorHandle], Set[ActorHandle]],
+        max_remote_requests_in_flight: int = 2,
+        ray_wait_timeout_s: Optional[float] = 0.03,
+        return_object_refs: bool = False,
+    ):
+        self._available_workers = workers if isinstance(workers, set) else set(workers)
+        self._unavailable_workers = set()
+        self._max_remote_requests_in_flight = max_remote_requests_in_flight
+        self._ray_wait_timeout_s = ray_wait_timeout_s
+        self._return_object_refs = return_object_refs
+        self._remote_requests_in_flight = defaultdict(set)
+        self._pending_to_actor = {}
+        self._call_queue = Queue(max_size=0)
+        self._pending_remotes = []
+
+    def submit(
+        self, remote_fn: Callable, fn_args: List[Any], fn_kwargs: Dict[str, Any]
+    ) -> None:
+        """Submit a remote function call to call on one of the workers
+
+        Args:
+            remote_fn: The remote function to call
+            fn_args: The arguments to pass to the remote function
+            fn_kwargs: The keyword arguments to pass to the remote function
+        """
+        call = (remote_fn, fn_args, fn_kwargs)
+        self._call_queue.put(call)
+
+    def run(self) -> int:
+        """Run submitted requests on available workers"""
```

Comment:
More detailed docstring?

Reply:
Its a private function now -- I'll probably leave as is.

Reply:
Hold on, I still would want to have the details here as a user:
Like for example: submit() just queues stuff w/o doing anything with the available workers, correct? Then run performs what? The actual remote call?